### PR TITLE
[AC-2521] Remove FlexibleCollectionsSignUp feature flag

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Remove the `FlexibleCollectionsSignUp` feature flag.

This was only ever used server-side to set `Organization` properties on creation, so it doesn't need to be persisted for any clients.

## Code changes

* set `Organization.FlexibleCollections` to `true`
* set `Organization.LimitCollectionCreationDeletion` to `false` (thereby enabling new FC behaviour) - this is done by actually removing the explicit assignment, which defaults to `false`
* remove references from test code, including an old test which will now never run. Update existing test to include this assertion

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Admin/AdminConsole/Models/OrganizationEditModel.cs [hunk 1]
```diff
@@ -162,19 +162,18 @@ public OrganizationEditModel(Provider provider)
...
             });
 
-    public Organization CreateOrganization(Provider provider, bool flexibleCollectionsSignupEnabled, bool flexibleCollectionsV1Enabled)
+    public Organization CreateOrganization(Provider provider, bool flexibleCollectionsV1Enabled)
     {
         BillingEmail = provider.BillingEmail;
 
         var newOrg = new Organization
         {
             // This feature flag indicates that new organizations should be automatically onboarded to
```
Comment (vincentsalucci) [inline src/Admin/AdminConsole/Models/OrganizationEditModel.cs:171]: Suggestion: give these comments an update to match the new actions (or lack thereof)

File: src/Core/AdminConsole/Services/Implementations/OrganizationService.cs [hunk 2]
```diff
@@ -575,12 +568,11 @@ public async Task<(Organization organization, OrganizationUser organizationUser,
 
             // This feature flag indicates that new organizations should be automatically onboarded to
```
Comment (vincentsalucci) [inline src/Core/AdminConsole/Services/Implementations/OrganizationService.cs:569]: Same as above.

File: src/Admin/AdminConsole/Models/OrganizationEditModel.cs (hunk 1 above)
Comment (eliykat) [reply src/Admin/AdminConsole/Models/OrganizationEditModel.cs]: I'm not sure exactly what you had in mind here but I updated the comments to be more specific, let me know what you think.

I also considered making the `Organization.FlexibleCollections` property read-only and always defaulting to `true`, but it's still used by the migration code which hasn't been deleted yet, so I maintained the current approach for now.

**Review by vincentsalucci — APPROVED** [review]
Thanks for the comment updates (and sorry for not providing more direction) I was just looking for them to be current with the code they're reflecting since the MVP features have been fully realeased, etc.

**Pull request merged**